File: 킬링캠프/1주 차 - 완전탐색, 백트래킹, 이분탐색/practice/Two Sum/Python/b. 재귀 풀이/Solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        return self.recursive(nums, target, 0, [])
        
    def recursive(self, nums, target, start, temp):
        if (len(temp) == 2):
            if (nums[temp[0]] + nums[temp[1]] == target):
                return temp
            return None

        for i in range(start, len(nums)):
            temp.append(i)

            answer = self.recursive(nums, target, i + 1, temp)
            if (answer != None):
                return answer
            
            temp.pop()

File: 킬링캠프/1주 차 - 완전탐색, 백트래킹, 이분탐색/practice/Two Sum/b. 재귀 풀이/Solution.java
import java.util.*;

class Solution {

    public int[] twoSum(int[] nums, int target) {
        return recursive(nums, target, 0, new ArrayList<Integer>());
    }

    private int[] recursive(int[] nums, int target, int start, List<Integer> answer) {
        if (answer.size() == 2) {
            if (nums[answer.get(0)] + nums[answer.get(1)] == target) {
                return new int[]{answer.get(0), answer.get(1)};
            }

            return null;
        }

        for (int i = start; i < nums.length; i++) {
            answer.add(i);
            int[] result = recursive(nums, target, i + 1, answer);
            if (result != null) {
                return result;
            }
            answer.remove(answer.size() - 1);
        }

        return null;
    }
}
